pkg/memcmdescriptor/provider.go

```diff
@@ -47,10 +45,25 @@ func New(cmDescriptorsFile io.Reader) (*Provider, error) {
 				"descriptors: %w", err)
 		}
 
-		if descriptors.InputDesc != nil {
-			presDef := presexch.PresentationDefinition{ID: uuid.NewString(), InputDescriptors: descriptors.InputDesc}
+		if descriptors.PresentationDefinition != nil {
+			var pd *presexch.PresentationDefinition
 
-			err := presDef.ValidateSchema()
+			presBytes, err := json.Marshal(descriptors.PresentationDefinition)
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal presentation "+
+					"definition: %w", err)
+			}
+
+			err = json.Unmarshal(presBytes, &pd)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unmarshal presentation"+
+					"definition: %w", err)
+			}
```

Comment:
Why do we need to marshal first and then unmarshal the same thing ?

Reply:
i have updated the logic to much simpler version , no longer require this